site/content/protocols/pickup/3.0/readme.md

```diff
@@ -0,0 +1,249 @@
+    "body": {
+        "message_id_list": ["123","456"]
+    }
+}
+```
+`message_id_list` is a list of `ids` of each message received. The `id` of each message is present in the attachment descriptor of each attached message of a delivery message.
+
+Upon receipt of this message, the `mediator` knows which messages have been received, and can remove them from the collection of queued messages with confidence. The `mediator` **SHOULD** send an updated `status` message reflecting the changes to the queue.
+
+#### Multiple Recipients
+
+If a message arrives at a `mediator` addressed to multiple `recipients`, the message **MUST** be queued for each `recipient` independently. If one of the addressed `recipients` retrieves a message and indicates it has been received, that message **MUST** still be held and then removed by the other addressed `recipients`.
+
+
+### Live Mode
+_Live Mode_ is the practice of delivering newly arriving messages directly to a connected `recipient`. It is disabled by default and only activated by the `recipient`. Messages that arrive when _Live Mode_ is off **MUST** be stored in the queue for retrieval as described above. If _Live Mode_ is active, and the connection is broken, a new inbound connection starts with _Live Mode_ disabled.
+
+Messages already in the queue are not affected by _Live Mode_; they must still be requested with `delivery-request` messages.
+
+_Live Mode_ **MUST** only be enabled when a persistent transport is used, such as WebSockets.
+
+Recipients have three modes of possible operation for message delivery with various abilities and level of development complexity:
+
+1. Never activate _Live Mode_. Poll for new messages with a `status_request` message, and retrieve them when available.
+2. Retrieve all messages from queue, and then activate _Live Mode_. This simplifies message processing logic in the `recipient`.
+3. Activate _Live Mode_ immediately upon connecting to the `mediator`. Retrieve messages from the queue as possible. When receiving a message delivered live, the queue may be queried for any waiting messages delivered to the same key for processing.
+### Live Mode Change
+_Live Mode_ is changed with a `live-delivery-change` message.
+
+Message Type URI: `https://didcomm.org/messagepickup/3.0/live-delivery-change`
+
+```json
+{
+    "id": "123456780",
+    "type": "https://didcomm.org/messagepickup/3.0/live-delivery-change",
+    "body": {
+        "live_delivery": true
+    }
+}
+```
+Upon receiving the `live_delivery_change` message, the `mediator` **MUST* respond with a `status` message.
+
+If sent with `live_delivery` set to true on a connection incapable of live delivery, a `problem_report` **SHOULD** be sent as follows:
+
+```json
+{
+  "id": "123456780",
+  "type": "https://didcomm.org/report-problem/2.0/problem-report",
+  "pthid": "< the value is the thid of the thread in which the problem occurred>",
+  "body": {
+        "code": "e.m.live-mode-not-supported",
```

Comment:
What does `e.m.` mean in this context?

Reply:
That was just an example using the [Problem Codes](https://identity.foundation/didcomm-messaging/spec/#problem-codes) defined in the spec. `e` is for error and `m` is when error was triggered by the previous message on the thread. I've never used this coding pattern, I'm expecting someone else to define it correctly.

Reply:
The `m` actually means that the error has message scope (that is, it is an indicator that the message has failed to achieve its intended purpose, but not that the protocol as a whole has failed).